ext/gvl_tracing_native_extension/gvl_tracing.c

```diff
@@ -75,25 +76,16 @@ static inline void initialize_thread_id(void) {
 }
 
 static inline void render_thread_metadata(void) {
-  uint64_t native_thread_id = 0;
-  #ifdef HAVE_GETTID
-    native_thread_id = gettid();
-  #elif HAVE_PTHREAD_THREADID_NP
-    pthread_threadid_np(pthread_self(), &native_thread_id);
-  #else
-    native_thread_id = current_thread_serial; // TODO: Better fallback for Windows?
-  #endif
-
   char native_thread_name_buffer[64] = "(unnamed)";
+  uint64_t thread_id = native_thread_id();
 
   #ifdef HAVE_PTHREAD_GETNAME_NP
     pthread_getname_np(pthread_self(), native_thread_name_buffer, sizeof(native_thread_name_buffer));
   #endif
 
   fprintf(output_file,
-    "  {\"ph\": \"M\", \"pid\": %u, \"tid\": %u, \"name\": \"thread_name\", \"args\": {\"name\": \"%lu %s\"}},\n",
-    process_id, current_thread_serial, native_thread_id, native_thread_name_buffer
-  );
+    "  {\"ph\": \"M\", \"pid\": %u, \"tid\": %llu, \"name\": \"thread_name\", \"args\": {\"name\": \"%llu %s\"}},\n",
+    process_id, thread_id, thread_id, native_thread_name_buffer);
 }
```

Comment:
The change to use `%llu` is actually not correct for Linux -- if you look at the CI runs, it complains that it's not correct of a uint64_t. (But it is correct for macOS -- going back to `%lu` makes the macOS build complain).

[I suggest using `PRIu64`](https://stackoverflow.com/questions/16859500/what-is-priu64-in-c), which should work correctly for both.

Reply:
I switched to `long int`. Maybe it's better?